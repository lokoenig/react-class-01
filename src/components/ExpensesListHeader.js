import React from "react";
import { expenseListTotal } from "../selectors/expenseListTotal";


export const ExpensesListHeader = (props) =>{
    const runningTotal = expenseListTotal(props);
    return (
        <>
        <h2>List of Expenses</h2>
        <div className="expense-list-header">
        <span>
            Description
        </span>
            <span className="total-amount">
            {new Intl.NumberFormat(
                'en-US',
                { style: 'currency', currency: 'USD' }
            ).format(runningTotal / 100)}
            </span>
            <span>
                Date Posted
            </span>
        </div>
        </>
    )
}

export default ExpensesListHeader;