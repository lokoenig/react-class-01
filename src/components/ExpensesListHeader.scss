.expense-list-header {
    &__qantity {
        font-weight: bold;
    }
}