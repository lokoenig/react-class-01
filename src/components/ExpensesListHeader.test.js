import React from "react";
import { render, screen } from '@testing-library/react';

import ExpensesListHeader from "./ExpensesListHeader";
import ExpenseTestData from "../fixtures/expenses";


test('snapshot header with standard fixture data', ()=>{

    const { getByText, asFragment } =
        render(
            <ExpensesListHeader
            expenses={ExpenseTestData} />
        );
    expect(asFragment()).toMatchSnapshot();
});