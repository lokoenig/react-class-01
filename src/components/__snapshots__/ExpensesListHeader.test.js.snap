// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot header with standard fixture data 1`] = `
<DocumentFragment>
  <h2>
    List of Expenses
  </h2>
  <div
    class="expense-list-header"
  >
    <span>
      Description
    </span>
    <span
      class="total-amount"
    >
      $18,036.32
    </span>
    <span>
      Date Posted
    </span>
  </div>
</DocumentFragment>
`;
